### ISSUE TITLE
Can't add manual receipe

### ISSUE THREAD

**Issue by rungis86** [nextcloud/cookbook#438]
**Description**
When I try to add a new receipe manually, when I click on the validate button, it turn and the receipe is not saved.

**Reproduction**
1. Click on "Créer une recette"
2. Fill the differents fields
3. Click on the check button

**Expected behavior**
No idea

**Screenshots**
![Screenshot 2020-12-15 at 10 25 14](https://user-images.githubusercontent.com/76039872/102196287-fe3cb880-3ebf-11eb-9953-fbd80fe020fd.png)

**Browser**
Firefox on chromebook via crostini or Firefox on Windows 10 or Google Chrome on chromebook

**Comment by TheMBeat** [comment]
Hello @rungis86 ,
can you try the Solution from [#430](https://github.com/nextcloud/cookbook/issues/430#issuecomment-744036064)?

**Comment by rungis86** [comment]
Thanks for your answer,
I have not the same problem with this version but there is an other bug with : I can add ingredient or ustensil but when I save the receipe, they disappear.

NC 20.0.3

**Comment by Johnsen92** [comment]
> Hello @rungis86 ,
> can you try the Solution from [#430](https://github.com/nextcloud/cookbook/issues/430#issuecomment-744036064)?

I had the same Issue as OP and the solution from [#430](https://github.com/nextcloud/cookbook/issues/430#issuecomment-744036064) fixed it, but now I have the same issue of disappearing ingredients when manually entered

**Comment by rungis86** [comment]
It's a little long but you can bypass this bug by manually editing the json file and add ingredients or instruction.

**Comment by christianlupus** [comment]
Could you make a further test, @rungis86 and @Johnsen92? Could you save the recipe then edit **after having saved** again and see if you can add now some tools/ingredients/instructions?

I can only confirm this behavior for newly added recipes. The data structure in Vue [here](https://github.com/nextcloud/cookbook/blob/master/src/components/EditInputField.vue#L6) is throwing an error if the recipe is not yet saved (and thus null). This might be solved by #386.

**Comment by Johnsen92** [comment]
As with the version from from [#430](https://github.com/nextcloud/cookbook/issues/430#issuecomment-744036064) creating a recipe, saving it and then editing it does not work either. What I tried: New recipe with some test title and test description -> Save -> Edit -> Save -> Edit and then I tried to put in some ingredients/steps/tools but the contents of the input fields still disappear after hitting enter or clicking "+ Add". It seems to be an issue with the way the form is handled on the client side, because if I, for example, add 4 blank ingredients, fill them with text and hit "Save changes", the json object sent in the PUT request to the server contains the array recipeIngredients with 4 empty string entries, so it seems the input fields for the ingredients are parsed incorrectly.

**Comment by Johnsen92** [comment]
> It's a little long but you can bypass this bug by manually editing the json file and add ingredients or instruction.

True, that would probably work, but I would also want for my mom and dad to be able to add new recipes and their JSON knowledge is a little rusty :)

**Comment by christianlupus** [comment]
OK, one more question to track this down: Can you do the following:

1. Create a new dummy recipe (arbitrary title) and save it.
2. Open the web developer tools (typically F12 or in the menu). There is the JS console. Empty it if there was some clutter using the small trash icon.
3. Reopen the recipe for editing.
4. Add the ingredients/tools/instructions
5. Check if any JS errors were reported so far
6. Save the recipe
7. Check for any further JS errors.

Thank you very much and a happy holiday and Merry Christmas!

**Comment by Johnsen92** [comment]
Except for these warnings I don't get any errors or outputs in the console:

"$ is deprecated: The global jQuery is deprecated. It will be updated to v3.x in Nextcloud 21. In later versions of Nextcloud it might be removed completely. Please ship your own."
"OC.Util.relativeModifiedDate is deprecated and will be removed in Nextcloud 21. See @nextcloud/moment"

Thank you for the great work, happy holidays and merry Christmas :)

**Comment by fkieber** [comment]
Same for me.

I can fill fields but when I click on adding something, nothing is happening.
Same when I want to save recipe. Nothing just the turning check button.

Solution from #430 give same résult. Ingredients and other are not saved.

**Comment by sebsgr** [comment]
I have the same problem at saving a receipt.
If I load the "new receip" Page there are many type errors in the js console:

`[Error] TypeError: null is not an object (evaluating 'this.recipe.totalTime="PT"+e+"H"+t+"M"') — RecipeEdit.vue:81`
`[Error] TypeError: null is not an object (evaluating 'e.$parent.recipe[e.fieldName]') — EditInputField.vue:83`

I get these errors an every textfield change at every recipe property too.  

When saving the recipe, I get these error on js console:
`TypeError: null is not an object (evaluating 'this.recipe.id')`

Editing a recipe works without any problems.

**Comment by christianlupus** [comment]
OK, in #386 this issue seems to be solved as a byproduct. The most recent build from the CI scripts is [here ATM](https://github.com/nextcloud/cookbook/actions/runs/437592200). (Later you might go directly via the PR's checks page at the bottom to get the newest one).
There is an artifact attached called `cookbook-appstore`. This is a `.zip` file of a `.tar.gz` that contains the corresponding dev version. If you are willing to test with it we can link this issue with the PR so the issue is closed once the PR is merged.

Please be warned that there is yet a small bug in the implementation at the time of writing that causes warnings in the JS logs and a minor regression in the UI, see the comments on the PR.

**Comment by lgnch** [comment]
A workaround while people are waiting for the fix:

1 paste a recipe in the web download link.
2 edit it and change its name
3 add your recipe and save.

Its not perfect, but for those who are willing to wait and run the latest version to assist with testing, this works fine.

**Comment by fkieber** [comment]
> A workaround while people are waiting for the fix:

Great. Simple but efficient ! Thanks.

**Comment by Johnsen92** [comment]
> OK, in #386 this issue seems to be solved as a byproduct. The most recent build from the CI scripts is [here ATM](https://github.com/nextcloud/cookbook/actions/runs/437592200). (Later you might go directly via the PR's checks page at the bottom to get the newest one).

I just tried it and it works, thanks a lot for your effort.

**Comment by MrReSc** [comment]
I am also affected by the same problem as the OP.

**Comment by christianlupus** [comment]
As #386 is merged just now, I am closing this issue here. If the problem persists (or is reappearing in a similar form), we can reopen or open a new issue.
